src/providers/TransactionBuilderContextProvider.tsx

```diff
@@ -715,33 +704,108 @@ const TransactionBuilderContextProvider = ({
       setTransactionBlockFieldValidationError,
       setCrossChainActionGasToken,
     }),
-    [],
+    []
   );
 
   const hideMenu = () => setShowMenu(false);
 
   const hasTransactionBlockAdded = transactionBlocks.some(
-    (transactionBlock) => transactionBlock.type === TRANSACTION_BLOCK_TYPE.ASSET_BRIDGE,
+    (transactionBlock) => transactionBlock.type === TRANSACTION_BLOCK_TYPE.ASSET_BRIDGE
   );
 
   const hasKlimaBlockAdded = transactionBlocks.some(
-    (transactionBlock) => transactionBlock.type === TRANSACTION_BLOCK_TYPE.KLIMA_STAKE,
+    (transactionBlock) => transactionBlock.type === TRANSACTION_BLOCK_TYPE.KLIMA_STAKE
   );
 
   const crossChainActionsInProcessing = useMemo(() => {
     if (!processingCrossChainActionIds?.length) return;
-    return dispatchedCrossChainActions?.filter(
-      (crossChainAction) => processingCrossChainActionIds.some((id) => id === crossChainAction.id),
+    return dispatchedCrossChainActions?.filter((crossChainAction) =>
+      processingCrossChainActionIds.some((id) => id === crossChainAction.id)
     );
   }, [processingCrossChainActionIds, dispatchedCrossChainActions]);
 
-  const hasProcessingUnsent = useMemo(() => !!getFirstCrossChainActionByStatus(
-    dispatchedCrossChainActions ?? [],
-    CROSS_CHAIN_ACTION_STATUS.UNSENT,
-  ), [dispatchedCrossChainActions]);
+  const hasProcessingUnsent = useMemo(
+    () => !!getFirstCrossChainActionByStatus(dispatchedCrossChainActions ?? [], CROSS_CHAIN_ACTION_STATUS.UNSENT),
+    [dispatchedCrossChainActions]
+  );
 
   const [showMulticallOptions, setShowMulticallOptions] = useState<string | null>(null);
 
+  const onBuyClick = async () => {
+    const maticSdk = getSdkForChainId(CHAIN_ID.POLYGON);
+
+    if (!accountAddress || !maticSdk) return;
```

Comment:
maybe consider adding any double click prevention (while deploying, etc)?

Reply:
Added event.preventDefault, and a check if deploying

Reply:
thx, also consider adding alert modal here too